src/Service/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -492,12 +494,12 @@ protected virtual DatabaseTable GenerateDbTable(string schemaName, string tableN
         /// <param name="schemaNames"></param>
         /// <param name="tableNames"></param>
         /// <returns>The dictionary populated with parameters.</returns>
-        protected virtual Dictionary<string, object?>
+        protected virtual Dictionary<string, Tuple<object?, DbType?>>
```

Comment:
Instead of this tuple, can we create a class representing such ConnectionParams (that has properties value and type)?